gradio/components.py

```diff
@@ -489,14 +498,21 @@ def set_interpret_parameters(
         self.interpretation_delta_type = delta_type
         return self
 
-    def get_interpretation_neighbors(self, x: float) -> Tuple[List[float], Dict]:
-        x = float(x)
+    def get_interpretation_neighbors(self, x: float | int) -> Tuple[List[float], Dict]:
+        x = int(x) if self.integer else float(x)
         if self.interpretation_delta_type == "percent":
             delta = 1.0 * self.interpretation_delta * x / 100
         elif self.interpretation_delta_type == "absolute":
             delta = self.interpretation_delta
         else:
             delta = self.interpretation_delta
+        if self.integer and math.floor(delta) != delta:
```

Comment:
I thought it was important to prevent users from trying to generate invalid neighbors if type="integer"